### DISCUSSION TITLE
Shared global.d.ts in monorepo

### DISCUSSION THREAD

**Discussion by moxley01** [vercel/vercel#8322] [Help]
Hi there,

I have a monorepo for a hobby project, and one of the top-level folders is a Next app. The folder structure is something like:

- root
    - next-app
            - package.json
            - tsconfig.json
    - backend
            - package.json
            - tsconfig.json
    - package.json
    - globals.d.ts

I want both `next-app` and `backend` to be able to use the type declarations defined in the root-level `globals.d.ts`. Currently I am trying to do this by having each tsconfig point outside itself i.e. 

```
{
  "include": ["**/*.ts", "../globals.d.ts"],
}
```

If I try to run `npm run build` in the `next-app` folder, everything is fine. However, if I run `vercel` in that same folder to deploy it, the build fails because it cannot find the types outside of the `next-app` folder.

Has anyone been able to successfully define typescript globals *outside* the NextJS folder that you are trying to deploy? Or is this just an artifact of using the Vercel CLI for deployment - would the git integration method of deployment succed here?

**Comment by amyegan** [comment]
Hi @moxley01. Are you using Turborepo, Nx, or another build system to manage your monorepo? 

If you can share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I would be happy to take a closer look.
